include/device_vector.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <vector>
+
+#include <quda_api.h>
+#include <malloc_quda.h>
+#include <util_quda.h>
+#include <complex_quda.h>
+
+/**
+   @file device_vector.h
+   Helper functionality for handling a small piece of device memory.
+ */
+
+namespace quda
+{
+  /**
+    @brief This class `device_vector` provides a light weighted wrapper for data on device.
+  */
+
+  template <class real> class device_vector
```

Comment:
A request for a global replace---I think this should be `DeviceVector`, not `device_vector`, to match otherwise ubiquitous naming conventions (`ColorSpinorField`, `GaugeField`, `SharedMemoryCache`, ...)

Reply:
Here it was originally meant to mimic `thrust::device_vector`. Are we sure we want to make this change?

Reply:
I think we should stick with `device_vector`

Reply:
Independent of the motivation I still think it's weird for it to be the one "memory container" that goes against our other style conventions, but I'll let it be.